packages/app/src/debug/DebugPageHeader.vue

```diff
@@ -0,0 +1,155 @@
+            <span class="sr-only">Dashboard Link:</span> View in the dashboard
+          </ExternalLink>
+        </li>
+      </ul>
+      <ul
+        :data-cy="'metadata'"
+        class="flex flex-wrap gap-x-2 items-center whitespace-nowrap children:flex children:items-center font-normal text-sm text-gray-700"
+      >
+        <li
+          :data-cy="'debug-results'"
+        >
+          <DebugResults
+            :gql="props.gql"
+          />
+        </li>
+        <li
+          v-if="debug?.commitInfo?.branch"
+          data-cy="debug-header-branch"
+        >
+          <i-cy-tech-branch-h_x16 class="mr-1 icon-dark-gray-300 mr-8px" />
+          <span class="sr-only">Branch Name:</span> {{ debug.commitInfo.branch }}
+        </li>
+        <li
+          v-if="props.commitHash"
+          data-cy="debug-header-commitHash"
+        >
+          <CommitIcon
+            class="fill-white h-16px w-16px mr-11px"
+          />
+          <span class="sr-only">Commit Hash:</span> {{ props.commitHash }}
+        </li>
+        <li
+          v-if="debug?.commitInfo?.authorName"
+          data-cy="debug-header-author"
+        >
+          <i-cy-general-user_x16
+            class="mr-1 icon-dark-gray-500 icon-light-gray-100 icon-secondary-light-gray-200 mr-11px"
+            data-cy="debug-header-avatar"
+          />
+          <span class="sr-only">Commit Author:</span> {{ debug.commitInfo.authorName }}
+        </li>
+        <li
+          v-if="debug.createdAt"
+          data-cy="debug-header-createdAt"
+        >
+          <IconTimeStopwatch
+            size="16"
+            class="mr-9px"
+            stroke-color="gray-500"
+            fill-color="gray-50"
+          />
+          <span class="sr-only">Run Total Duration:</span> {{ debug.totalDuration }} ({{ relativeCreatedAt }})
+        </li>
+      </ul>
+    </div>
+  </div>
+</template>
+<script lang="ts" setup>
+import { computed } from 'vue'
+import DebugResults from './DebugResults.vue'
+import ExternalLink from '@cy/gql-components/ExternalLink.vue'
+import type { DebugPageFragment } from '../generated/graphql'
+import CommitIcon from '~icons/cy/commit_x14'
+import { IconTimeStopwatch } from '@cypress-design/vue-icon'
+import { gql } from '@urql/core'
+import { dayjs } from '../runs/utils/day.js'
+
+// runNumber and commitHash dont currently exist in the query and therefore are being obtained as props instead
+gql`
+fragment DebugPage on CloudRun {
+	id
+	createdAt
+	status
+  totalDuration
+  url
+  tags {
+    id
+    name
+  }
+	...RunResults
+	commitInfo {
+		authorName
+		authorEmail
+		summary
+		branch
+	}
+}
+`
+
+const props = defineProps<{
+  gql: DebugPageFragment
+  commitsAhead: string
+  commitHash: string
+  runNumber: number
+}>()
+
+const debug = computed(() => props.gql)
+
+const relativeCreatedAt = computed(() => dayjs(new Date(debug.value.createdAt!)).fromNow())
+
+</script>
+<style scoped>
+ul:nth-child(2) li:not(:first-child)::before {
```

Comment:
@amehta265 What Lachlan is referring to here is the `ul:nth-child(2)` reference.  This is specifically targeting the second `ul` in the component, but it would break if someone added another `ul` ahead of it.  You could add a class to that `ul` and target that with your CSS selector instead of `ul-nth-child(2)`

Reply:
I see, that makes a lot of sense. I will make the necessary updates